### PR TITLE
Add SlicerSPoM extension (providing a 3D Poisson equation based shape analysis module)

### PR DESCRIPTION
**Pull request by gaoyi**
# New extension

- [x] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [x] Repository name is Slicer+ExtensionName
- [x] Repository is associated with `3d-slicer-extension` GitHub topic so that it is listed [here](https://github.com/topics/3d-slicer-extension). To edit topics, click the settings icon in the right side of "About" section header and enter `3d-slicer-extension` in "Topics" and click "Save changes". To learn more about topics, read https://help.github.com/en/articles/about-topics
- [x] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [x] Any known related patents must be mentioned in the extension description.
- [x] If source code license is more restrictive for users than BSD, MIT, Apache, or 3D Slicer license then the name of the used license must be mentioned in the extension description. The source code depends on packages GSL and alglib, which are GPL type license.
- [x] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (main, master, release, ...) instead of a specific git has to avoid re-submitting pull request whenever the extension is updated
- [x] Extension icon URL is correct (do not use the icon's webpage but the raw data download URL that you get from the download button - it should look something like this: https://raw.githubusercontent.com/user/repo/main/SomeIcon.png)
- [x] Screenshot URLs (screenshoturls) are correct, contains at least one
- [x] Homepage URL points to valid webpage containing the following:
  - [x] Extension name
  - [x] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [x] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [x] Description of contained modules: at one sentence for each module
  - [ ] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set. Tutorial working in progress
  - [x] Publication: link to publication and/or to PubMed reference (if available)
  - [x] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/master/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work. Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.
  - [x] Content of submitted s4ext file is consistent with the top-level CMakeLists.txt file in the repository (description, URLs, dependencies, etc. are the same)

### CODE REVIEW

File: SlicerSPoM.s4ext [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# list dependencies
+# - These should be names of other modules that have .s4ext files
+# - The dependencies will be built first
+depends NA
+
+# Inner build directory (default is ".")
+build_subdirectory .
+
+# homepage
+homepage https://github.com/gaoyi/SlicerSPoM
```
Comment (pieper) [inline SlicerSPoM.s4ext:21]: I don't see a license file or information about how to use this extension in https://github.com/gaoyi/SlicerSPoM

Comment (gaoyi) [reply SlicerSPoM.s4ext:21]: Thanks Steve, This is added. I copied Slicer's License file to the repository home folder.

**Comment by jcfr** [comment]
Thanks for contributing extensions :pray: 

To streamline future maintenance and help prospective users, here are the follow-up comment.

## SlicerSPoM

* **Important** The code found in the `alglib` folder is licensed using GPL2 or any later version:

  > ALGLIB Free Edition</b> is distributed under GPL 2+, GPL license version 2 or at your option any later version.
  > 
  > Source: https://raw.githubusercontent.com/gaoyi/SlicerSPoM/main/PoissonShapeAnalysis/lib/alglib/manual.cpp.html

  While the Slice license (similar to the BSD 3-clause) is likely compatible with GPLv3 [^1], the compiled extensions along side the Slicer install where is installed will also be licensed under the GPLv3 license and would require source code all its components to be available. This make the use of `SlicerSPoM` impossible in the context of commercial projects.

  I would then suggest to revisit the use of `alglib`.

[^1] https://www.gnu.org/licenses/license-list.html#GPLCompatibleLicenses

* Consider renaming the extension to have a more explicit name. For example, `SlicerPoissonShapeAnalysis`. That would also be consistent with the name of the associated CLI module. Tracked by issue https://github.com/gaoyi/SlicerSPoM/issues/1

* Consider reorganizing the [README.md](https://github.com/gaoyi/SlicerSPoM/blob/main/README.md). Indeed, the current organization assumes the reader is already familiar with the approach but we can't assume it is the case for broader dissemination. Suggested layout and example of project documented in issue https://github.com/gaoyi/SlicerSPoM/issues/2

* Attempting to configure the project fails due to missing GSL headers. This is tracked by issue https://github.com/gaoyi/SlicerSPoM/issues/3

* Update baseline and input test datasets to valid ones. This is tracked by https://github.com/gaoyi/SlicerSPoM/issues/4

* Fix `-Wunused-parameter` and `-Wmaybe-uninitialized` warnings. See https://github.com/gaoyi/SlicerSPoM/issues/5 and https://github.com/gaoyi/SlicerSPoM/issues/6

* Remove commented code. See https://github.com/gaoyi/SlicerSPoM/issues/7

* Simplify build-system. Corresponding changes proposed in https://github.com/gaoyi/SlicerSPoM/pull/10 which should also address https://github.com/gaoyi/SlicerSPoM/issues/3

* Remove use of deprecated ITK function using ITKv5 interface. See https://github.com/gaoyi/SlicerSPoM/issues/9 (it contains link to migration guide)

**Review by jcfr — CHANGES_REQUESTED** [review]
Reported issues should be reviewed before we can move forward.

**Pull request closed without merging**